src/Device/Driver/FANET/Parser.cpp

```diff
@@ -0,0 +1,304 @@
+
+  // Skip extended header
+  if(IsNthBitSet(payload[0], FANET_SERVICE_BIT_EHEADER))
+  {
+    // TODO: decode extended header byte if set, currently not needed/used
+    payload_pos++;
+  }
+
+  // Return if no location
+  if(payload_pos + 6 > payload_length)
+    return false;
+
+  station->location = Payload2GeoPoint(&payload[payload_pos]);
+  if( ! station->location.IsValid())
+    return false;
+  station->location_available = true;
+  payload_pos += 6;
+
+  /*
+   * Temperature (+1byte in 0.5 degree, 2-Complement)
+   */
+  if(IsNthBitSet(payload[0], FANET_SERVICE_BIT_TEMP))
+  {
+    if(payload_pos + 1 > payload_length)
+      return false;
+    station->temperature = (float)payload[payload_pos++] / 2.0f;
+  }
+
+  /*
+   * 	Wind (+3byte: 1byte Heading in 360/256 degree, 1byte speed and 1byte
+	 *  gusts in 0.2km/h (each: bit 7 scale 5x or 1x, bit 0-6))
+   */
+  if(IsNthBitSet(payload[0], FANET_SERVICE_BIT_WIND))
+  {
+    if(payload_pos + 3 > payload_length)
+      return false;
+
+    station->wind_dir_deg = (((float)payload[payload_pos++]) / 0.7083333334f);
+    station->wind_speed_kmph =
+        Payload2Float(payload[payload_pos++], 5.0f) * 0.2f;
+    station->wind_gust_kmph = Payload2Float(payload[payload_pos++], 5.0f) * 0.2f;
+  }
+
+  /*
+   * Humidity (+1byte: in 0.4% (%rh*10/4))
+   */
+  if(IsNthBitSet(payload[0], FANET_SERVICE_BIT_HUMIDITY))
+  {
+    if(payload_pos + 1 > payload_length)
+      return false;
+    station->rel_humidity = payload[payload_pos++] / 2.5f;
+  }
+
+  /*
+   * Barometric pressure normailized (+2byte: in 10Pa, offset by 430hPa,
+   * unsigned little endian (hPa-430)*10)
+   */
+  if(IsNthBitSet(payload[0], FANET_SERVICE_BIT_PRESSURE))
+  {
+    if(payload_pos + 2 > payload_length)
+      return false;
+    int lsb = payload[payload_pos++];
+    int msb = payload[payload_pos++];
+    station->preasure_hpa = (((msb << 8) | lsb) / 10.0) + 430.0;
+  }
+
+  /*
+   * State of Charge  +1byte
+   * (lower 4 bits: 0x00 = 0%, 0x01 = 6.666%, .. 0x0F = 100%)
+   */
+  if(IsNthBitSet(payload[0], FANET_SERVICE_BIT_SOC))
+  {
+    if(payload_pos + 1 > payload_length)
+      return false;
+    station->soc_percent =
+        Payload2Float(payload[payload_pos++], 1.0f) * 100.0f / 15.0f;
+  }
+
+  return true;
+}
+
+/*
+ *  Service(type 4 - weather info):
+ *  #FNF src_manufacturer,src_id,broadcast,signature,type,payloadlength,payload
+ */
+bool
+FanetDevice::ParseRxPacket(char *cmd_str, StationsList &stations_list,
+                           double clock) {
+  // Prepare the line
+  cmd_str = RemoveExtraChars(cmd_str);
+  if ( ! IntegrityCheck(cmd_str))
+    return false;
+
+  char *start_ptr = (char *)cmd_str;
+  char *end_ptr;
+
+  if(! start_ptr)
+    return false;
+
+  // Manufacturer(1 byte)
+  uint8_t manufacturer = strtol(start_ptr, &end_ptr, 16);
+  if(start_ptr == end_ptr)
+    return false;
+
+  start_ptr = strchr(start_ptr, ',') + 1;  // comma after src_manufacturer
+  if (start_ptr == nullptr)
```

Comment:
If there's no comma, then `start_ptr` will *not* be `nullptr`, but `nullptr+1`. This your code will still crash due to `nullptr` dereference!
Also I don't like this way of parsing at all. It looks very clumsy. You called `strtol()`, and you know where that hex number ends - at `end_ptr`. Still you start looking for the comma at `start_ptr`.